File: apps/api/src/main/java/com/rpg/redsunapi/user/dto/UserSettingsDto.java
package com.rpg.redsunapi.user.dto;

import com.rpg.redsunapi.user.UserSettings;

public record UserSettingsDto(
  String appLanguage,
  String appTheme,
  boolean redirectToFavorite
) {
  public static UserSettingsDto from(UserSettings settings) {
    return new UserSettingsDto(
      settings.getAppLanguage().name(),
      settings.getAppTheme().getValue(),
      settings.isRedirectToFavorite()
    );
  }
}

File: apps/web/src/interface/dtos/user/UserSettingsDTO.ts
import {ELanguage} from "../../enums/ELanguage";
import {EThemeApplication} from "../../enums/EThemeApplication";

export interface UserSettingsDTO {
  appLanguage: ELanguage;
  appTheme: EThemeApplication;
  redirectToFavorite: boolean;
  favoriteTaleId: string | null;
}
